src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -288,7 +288,7 @@ public void endPatientVisit() {
 
 	@Then("the system ends the patient visit")
 	public void systemEndsPatientVisit() {
-		assertNull(visitsDashboardPage.getActiveVisit());
+		assertPage(visitsDashboardPage.waitForPage());
```

Comment:
I think we are supposed to assert that the patient visit is ended successfully. Probably the previous [step](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature#L43) that is implemented [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a8887b1076faaa0f1dd136f53a5556dc0406f44/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java#L108-L113) that may have a flaw that is causing the next step to fail.

Reply:
This  error is from that line 291, its asserts activeVisit when actually there is no active visit. thats why its fails.with java.lang.assertion error Exception.

Reply:
do you mind sharing the logs?

Reply:
https://pastebin.com/S1Qumw7K  , You can test it out locally to confirm.

Reply:
Have skimmed through the logs and am skeptical whether we need to wait for page [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a8887b1076faaa0f1dd136f53a5556dc0406f44/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java#L112) after the assertion of the visit note. Isn't that a flaw that may be causing the issue?

Reply:
you can re read through the logs, this commit fixes the issue , According to the logs showing line 291 is failing due to . java.lang.AssertionError: expected null

Reply:
AFAICT some the test may have a flaw and points the logs at another point, have you looked into the issue [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/263#discussion_r764577316)?

Reply:
What could be your opinion of how to fix this issue according to the logs, the fact that this passes on both firefox and chrome seems to have fix the issue, Looking forward to see your opinion.

Reply:
well the goal is to ensure the test does not break on ci and have often seen the workflow failing in Bamboo. Have you looked at whether we need this method [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a8887b1076faaa0f1dd136f53a5556dc0406f44/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java#L112) within the workflow?

Reply:
i think that method doesnt have any harm to the build, let me remove that.

Reply:
Here are the [logs](https://github.com/openmrs/openmrs-contrib-qaframework/runs/4453734422?check_suite_focus=true) citing out the [guy](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a8887b1076faaa0f1dd136f53a5556dc0406f44/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java#L112).

Reply:
> This error is from that line 291, its asserts activeVisit when actually there is no active visit. thats why its fails.with java.lang.assertion error Exception.

@sherrif10 isn't it an assertNull() method. It actually asserts that active visits are not present.

Reply:
you can recheck my last comment.

Reply:
What could be the use of [this line here ](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a8887b1076faaa0f1dd136f53a5556dc0406f44/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java#L86) as per the user story. I think we need to clear up some redundant code from the steps. 
@kdaud  what are your thoughts?